marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.sjs

```diff
@@ -49,19 +51,31 @@ function getTypesInfo() {
 }
 
 function getArtifacts(artifactType) {
-    const artifacts = [];
     const queriesForAnd = [];
     const artifactLibrary =  getArtifactTypeLibrary(artifactType);
     for (const coll of artifactLibrary.getCollections()) {
         queriesForAnd.push(cts.collectionQuery(coll));
     }
     if (queriesForAnd.length) {
-        const results = cts.search(queriesForAnd.length > 1 ? cts.andQuery(queriesForAnd) : queriesForAnd[0]);
-        for (const result of results) {
-            artifacts.push(result.toObject());
+        if (artifactType == "loadData") {
+            return cts.search(queriesForAnd.length > 1 ? cts.andQuery(queriesForAnd) : queriesForAnd[0]).toArray();
```

Comment:
One nit - I would just call this "queries" (the plural implying that it's an array). And then it doesn't matter if there's one or more - you can just do:

    cts.search(cts.andQuery(queries)).toArray()

Reply:
Why do you think CollectionsController/Manager should be removed? I noticed that we could not reuse CollectionsService in com.marklogic.hub.web.service package and also depends on HubConfig.